generate.py

```diff
@@ -40,7 +40,9 @@ def generate(
     T_new = T + max_new_tokens
     if max_seq_length is None:
         max_seq_length = min(T_new, model.config.block_size) 
-
+        #Assertion as an option it works during inference, undesirable during training
+        assert max_seq_length <= T_new, f"Prevent allocate unnecessary memory: max_seq_length: {max_seq_length} <= T_new: {T_new}"
```

Comment:
This isn't quite what I was thinking. What you had in https://github.com/Lightning-AI/lit-parrot/pull/71 was closer. That is

`main` gets a `max_new_tokens` argument
`generate` gets a `T_new` argument
This assertion happens before the `generate` call.

What do you think?

Reply:
In the previous solution there was a problem when I removed max_new_tokens from generate() function which took new argument T_new computed in main(), I had to calculate max_new_tokens from T_new in generate().  The idea was that during training generate() could return longer sequences and to prevent this during inference, it was done by providing the argument T_new or not during validation, and here there was a problem with what to calculate max_new_tokens.  I messed up the code a bit and got lost.  That's why it seemed to me that the current solution is quite simple and elegant, it prevents the assertion from occurring during the training and it should occur during the inference because we don't provide the max_seq_lenght argument then, only in this case it is calculated inside generate() and then the assertion should work.  If the previous solution was better, we can go back to it

Reply:
I see. Thanks for your explanation. I think using `max_seq_length` for this isn't right  because it could be passed during generation too. This solution is simpler to implement but maybe not that straightforward to understand